dynawo/sources/Solvers/Common/DYNSolverFactory.h

```diff
@@ -60,13 +61,15 @@ class SolverFactory {
    */
   virtual void destroy(Solver*) const = 0;
 
+  using SolverPtr = std::unique_ptr<Solver, std::function<void(Solver*)> >;
```

Comment:
second template argument is supposed to be a deleter I am not sure this is the case here

Reply:
The deleted is added in dynawo/sources/Solvers/Common/DYNSolverFactory.cpp :

`solverPtr = SolverPtr(solver, deleteSolver);`